File: Social_medie_projekt/Client/src/app/_models/post.ts
import { Likes } from "./likes";
import { Tag } from "./tags";
import { User } from "./user"

export interface Post {
    userId?: number;
    postId: number;
    title: string;
    desc: string;
    likes?: number;
    date?: Date;
    user?: User;
    tags?: Tag[];
}

File: Social_medie_projekt/WebApi/Database/Entities/Post.cs
﻿namespace WebApi.Database.Entities
{
    public class Post
    {
        [Key]
        public int PostId { get; set; }

        [Column(TypeName = "nvarchar(100)")]
        public string Title { get; set; } = string.Empty;

        [Column(TypeName = "nvarchar(1000)")]
        public string Desc { get; set; } = string.Empty;

        [Column(TypeName = "int")]
        public int Likes { get; set; } = 0;

        [Column(TypeName = "datetime")]
        public DateTime Date { get; set; } = DateTime.Now;



        [ForeignKey("User.UserId")]
        public int UserId { get; set; }

        public User User { get; set; }
    }
}
